service.go

```diff
@@ -1,186 +1,59 @@
 package main
 
 import (
-	"errors"
 	"fmt"
 	"github.com/go-playground/validator/v10"
-	"strconv"
-	"time"
 )
 
-type RawInput struct {
-	Id       string `json:"id" validate:"gte=0,numeric"`
-	Name     string `json:"name" validate:"required,min=2,max=30"`
-	Interval string `json:"interval" validate:"required,gt=0,numeric"`
+type DevicePayload struct {
+	Name     string  `json:"name" validate:"required,min=2,max=30"`
+	Interval int     `json:"interval" validate:"gt=0,numeric"`
+	Value    float64 `json:"value" validate:"numeric"`
 }
 
-type ParsedInput struct {
-	Id       int
-	Name     string
-	Interval int
+type DeviceDao interface {
+	AddDevice(device *DevicePayload) (int, error)
 }
 
 type Service struct {
-	devices         map[int]Device
-	DevicesReadings *DataAccessObject
-	DevicesSaveChan chan DeviceInfo
-	stopChan        chan bool
-}
-
-func (s *Service) init(dao *DataAccessObject) {
-	s.DevicesReadings = dao
-	s.DevicesSaveChan = make(chan DeviceInfo)
-	s.stopChan = make(chan bool)
-	s.devices = make(map[int]Device)
+	Dao       DeviceDao
+	validator *validator.Validate
 }
 
 func (s *Service) run() {
-	go s.tickerService()
+	s.validator = validator.New()
```

Comment:
Run is private - it will be a problem in case of refactoring to call in from main. Also, no one outside of this service should call private method.

This method does only initialisation. We should keep usage of a service as simple and as obvious as possible, even without looking at its code or all methods.
In golang, for initialisation usually you have two options:
- initialise struct directly
- initialise using constructor/factory func

I would use the later option. Provide something like `func NewService() *Service)` that handles initialisation for the user of the service/